terraform/main.tf

```diff
@@ -0,0 +1,61 @@
+# Enable Google Cloud APIs
+module "enable_google_apis" {
+  source                      = "terraform-google-modules/project-factory/google//modules/project_services"
+  version                     = "~> 13.0"
+
+  project_id                  = var.gcp_project_id
+  disable_services_on_destroy = false
+
+  activate_apis               = var.apis
+}
+
+# Create GKE Cluster
+resource "google_container_cluster" "cluster" {
+  name               = var.name
```

Comment:
nit: i would advise to auto-generate the name. we should leave minimum work to the user and figuring out a unique name for the cluster in the existing project can be challenging. you can use `random` provider to generate a random suffix for the predefined name.